Shopify/cli
# Refresh the theme development server session every 30 minutes

### WHY are these changes introduced?

The remote renderer was frequently authenticating, which impacted the performance of the development server. Now, the development server session is automatically refreshed every 30 minutes.

With this change, the rendering requests become much faster:

**Before**
![Screenshot 2024-08-30 at 13 06 09](https://github.com/user-attachments/assets/59207ab9-1e39-486d-ab02-89ab5a85f435)

**After**
![Screenshot 2024-08-30 at 13 05 32](https://github.com/user-attachments/assets/5a040fb0-276b-4e5d-a37c-abf52445a7ac)

### WHAT is this pull request doing?

This PR extracts the `DevServerSession` creation to the `dev-server-session.ts` module, which creates and refreshes the session.

### How to test your changes?

- Update the `SESSION_TIMEOUT_IN_MS` to 10 seconds (`10000`) [here](https://github.com/Shopify/cli/blob/03e53f9d4fd929b346c61e96a5f76da04e6ff06d/packages/theme/src/cli/utilities/theme-environment/dev-server-session.ts#L8)
- Run `shopify theme dev --dev-preview --verbose`
- Notice that the dev session is updated every 10 seconds and that rendering now takes half the time 🚀

### Post-release steps

N/A

### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
